### ISSUE TITLE
Problems with ImageMagick 7.0.2-0 and imagick 3.4.1

### ISSUE THREAD

**Issue by jacksbox** [Intervention/image#607]
I´m using 
`imagick 3.4.1 compiled`
with
`ImageMagick 7.0.2-0 Q16 i686 2016-06-13 http://www.imagemagick.org`
on my server.

this leads to following error:

```
FatalThrowableError in Decoder.php line 23:
Undefined class constant 'IMGTYPE_TRUECOLORMATTE'
```

It´s possible to supres this error by replacing 'IMGTYPE_TRUECOLORMATTE' with 7 (the value IMGTYPE_TRUECOLORMATTE should provide, i guess).

But than still I get a problem when working with masks: They get not applied: The mask image is loaded, but when applied with ->mask(), nothing happens. The original image stays the same.
If I take the imagemagick instance and apply the mask via the normal imagemagick command, everything works fine.

Any ideas, thoughts, solutions?

**Comment by nyon** [comment]
I had the same problem and fixed it by replacing `IMGTYPE_TRUECOLORMATTE` by `IMGTYPE_TRUECOLOR`. [This](http://stackoverflow.com/a/13557199/2471991) post on stackoverflow says

> instead of the formerly imagick::IMGTYPE_TRUECOLORMATTE

So I think this package is using an old constant leading to that problem?

Perhaps checking if `IMGTYPE_TRUECOLORMATTE` is a defined constant would be a viable solution.

I do not use masks, so I don't know if this also fixes your problem.

**Comment by jorenvanhee** [comment]
@nyon Thanks 👍  Worked for me.

**Comment by valentinxxx** [comment]
Fixed in #679, released in 2.3.11
